### ISSUE TITLE
[Enhancement] More Accessible Font Sizes

### ISSUE THREAD

**Issue by rachelkang** [dotnet/maui#831]
# More Accessible Font Sizes

There are a number of improvements to be made with regard to ensuring accessible font sizes. Being able to support a wide range of font sizes, especially with various accessibility settings for font scaling, is key.

This issue will be updated as exploration continues.

## Named fonts

[NamedSize](https://docs.microsoft.com/dotnet/api/xamarin.forms.namedsize?view=xamarin-forms) and the [corresponding platform-specific sizes](https://docs.microsoft.com/xamarin/xamarin-forms/user-interface/text/fonts#understand-named-font-sizes) should be reinvestigated. 

Considerations, among others, should include:
- Minimum font size requirements for accessibility
- Consistency across platforms vs. with native platforms

Related issues:
- https://github.com/xamarin/Xamarin.Forms/issues/13628
- https://github.com/xamarin/Xamarin.Forms/issues/13543

## Dynamic font sizes

Text should be rendered fully and appropriately, (without getting cut short anywhere), despite the font size, on all different...
- Platforms and devices
- Layouts and screen sizes
- Control sizes

This includes interaction with platform/device-specific accessibility settings for font scaling. For example, if the font is scaled to 200% on a device, all text in .NET MAUI apps should be scaled and rendered fully and appropriately.

Related issues:
- https://github.com/xamarin/Xamarin.Forms/issues/13448
- https://github.com/xamarin/Xamarin.Forms/issues/11270

**Comment by GalaxiaGuy** [comment]
I am pleased these things are getting more attention :)

There are few extra features around this I would like support for (some of these are Forms-y and some Essentials-y but I guess they all go here now?):

- Text that isn't scaled, regardless of device setting.
- Scale an arbitrary size in the same way as font size.

(Of the two, I've used the second more often - think something like sizing the image of a custom radio button).

- Query the current accessible font scaling details.
- A notification when it changes.
